### ISSUE TITLE
bug: patch selection malfunctions

### ISSUE THREAD

**Issue by oSumAtrIX** [revanced/revanced-manager#543]
### Type

Error while running the manager

### Bug description

If two patches have the same name, but are constrained to different packages, the ReVanced manager might still get confused and include the wrong patch.

RE: https://github.com/revanced/revanced-patches/issues/1122#issuecomment-1328083744

### Steps to reproduce

From https://github.com/revanced/revanced-patches/issues/1122:

> 1. Install windy.app from Google play, or download from sources like apkpure
> 2. Patch with manager 0.0.40 with unlock-pro patch
> 3. notice there are multie errors
> 4. repeat steps with manager 0.0.41

### Manager version

0.0.40

### Target package name

windy.app

### Installation type

Non-root

### Patches selected.

- unlock-pro

### Device logs (exported using Manager settings).

```shell
11-27 00:55:11.611 19080 28331 W System.err: app.revanced.patcher.patch.PatchResultError
11-27 00:55:11.611 19080 28331 W System.err: 	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:346)
11-27 00:55:11.611 19080 28331 W System.err: 	at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:37)
11-27 00:55:11.611 19080 28331 W System.err: 	at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:362)
11-27 00:55:11.611 19080 28331 W System.err: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
11-27 00:55:11.611 19080 28331 W System.err: 	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:129)
11-27 00:55:11.611 19080 28331 W System.err: 	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda$19(MainActivity.kt:328)
11-27 00:55:11.611 19080 28331 W System.err: 	at app.revanced.manager.flutter.MainActivity.$r8$lambda$E2jhTFrLKJQp09EHxMNmoL-t4AE(Unknown Source:0)
11-27 00:55:11.611 19080 28331 W System.err: 	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda4.run(Unknown Source:24)
11-27 00:55:11.611 19080 28331 W System.err: 	at java.lang.Thread.run(Thread.java:1012)
11-27 00:55:11.611 19080 28331 W System.err: Caused by: java.lang.NullPointerException
11-27 00:55:11.611 19080 28331 W System.err: 	at app.revanced.patches.iconpackstudio.misc.pro.patch.UnlockProPatch.execute(UnlockProPatch.kt:27)
11-27 00:55:11.611 19080 28331 W System.err: 	at app.revanced.patches.iconpackstudio.misc.pro.patch.UnlockProPatch.execute(UnlockProPatch.kt:16)
11-27 00:55:11.611 19080 28331 W System.err: 	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:342)
11-27 00:55:11.611 19080 28331 W System.err: 	... 8 more
11-27 00:55:11.611 19080 28331 W System.err: app.revanced.patcher.patch.PatchResultError: 'unlock-pro' did not succeed previously
11-27 00:55:11.611 19080 28331 W System.err: 	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:300)
11-27 00:55:11.611 19080 28331 W System.err: 	at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:37)
11-27 00:55:11.611 19080 28331 W System.err: 	at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:362)
11-27 00:55:11.612 19080 28331 W System.err: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
11-27 00:55:11.612 19080 28331 W System.err: 	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:129)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda$19(MainActivity.kt:328)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.manager.flutter.MainActivity.$r8$lambda$E2jhTFrLKJQp09EHxMNmoL-t4AE(Unknown Source:0)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda4.run(Unknown Source:24)
11-27 00:55:11.612 19080 28331 W System.err: 	at java.lang.Thread.run(Thread.java:1012)
11-27 00:55:11.612 19080 28331 W System.err: app.revanced.patcher.patch.PatchResultError: 'unlock-pro' did not succeed previously
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:300)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:37)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:362)
11-27 00:55:11.612 19080 28331 W System.err: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
11-27 00:55:11.612 19080 28331 W System.err: 	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:129)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda$19(MainActivity.kt:328)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.manager.flutter.MainActivity.$r8$lambda$E2jhTFrLKJQp09EHxMNmoL-t4AE(Unknown Source:0)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda4.run(Unknown Source:24)
11-27 00:55:11.612 19080 28331 W System.err: 	at java.lang.Thread.run(Thread.java:1012)
11-27 00:55:11.612 19080 28331 W System.err: app.revanced.patcher.patch.PatchResultError: 'unlock-pro' did not succeed previously
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:300)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:37)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:362)
11-27 00:55:11.612 19080 28331 W System.err: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
11-27 00:55:11.612 19080 28331 W System.err: 	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:129)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda$19(MainActivity.kt:328)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.manager.flutter.MainActivity.$r8$lambda$E2jhTFrLKJQp09EHxMNmoL-t4AE(Unknown Source:0)
11-27 00:55:11.612 19080 28331 W System.err: 	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda4.run(Unknown Source:24)
11-27 00:55:11.612 19080 28331 W System.err: 	at java.lang.Thread.run(Thread.java:1012)
```

### Screenshots or videos

![image](https://user-images.githubusercontent.com/13122796/204101448-8c6183d8-791b-4a50-a6cf-40e90097eab4.png)

### Solution

The ReVanced Manager has to filter the patches based on the selected app package and then filter on the selected patches.

### Additional context

The CLI does not have this issue.

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.
- [X] The issue is related solely to the ReVanced Manager

**Comment by FineFindus** [comment]
Since the issue is the patch name, this would also include `nyx` (`iconpackstudio`) and `myexpenses`, not just `windy.app`, since all have a `unlock-pro` named patch

**Comment by Ushie** [comment]
Fixed in 922f474b59bb28a095336a0c0947e94d8bc5edb2
